src/main/java/dev/morling/jfrunit/JfrEventTestSpockExtension.java

```diff
@@ -0,0 +1,104 @@
+/**
+ *  Copyright 2020 - 2021 The JfrUnit authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package dev.morling.jfrunit;
+
+import org.spockframework.runtime.extension.AbstractAnnotationDrivenExtension;
+import org.spockframework.runtime.extension.AbstractMethodInterceptor;
+import org.spockframework.runtime.extension.IMethodInvocation;
+import org.spockframework.runtime.model.FieldInfo;
+import org.spockframework.runtime.model.SpecInfo;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class JfrEventTestSpockExtension extends AbstractAnnotationDrivenExtension<JfrEventTest> {
+
+    @Override
+    public void visitSpecAnnotation(JfrEventTest annotation, SpecInfo spec) {
+        final JfrMethodInterceptor interceptor = new JfrMethodInterceptor();
+        spec.getAllFeatures().stream().forEach(featureInfo -> {
```

Comment:
Depending on how you want to deal with inheritance, you might want to call `spec.getBottomSpec().getAllFeatures()` instead.

Reply:
Thanks for your suggestion!